.infrastructure/helm-chart/todoapp/templates/app/clusterIp.yml

```diff
@@ -0,0 +1,15 @@
+apiVersion: v1
+kind: Service
+metadata:
+  name: {{ .Chart.Name }}-service
+  namespace: {{ .Values.common.namespace }}
+spec:
+  type: ClusterIP
+  selector:
+    app: {{ .Values.todoapp.labelName }}
```

Comment:
Ensure that `.Values.todoapp.labelName` is defined in your `values.yaml` file. This is necessary for proper label matching.